sources/packages/backend/apps/api/src/route-controllers/student/student.controller.service.ts

```diff
@@ -64,25 +71,57 @@ export class StudentControllerService {
     groupName: string,
     auditUserId: number,
   ): Promise<FileCreateAPIOutDTO> {
-    const createdFile = await this.fileService.createFile(
-      {
-        fileName: file.originalname,
-        uniqueFileName: uniqueFileName,
-        groupName: groupName,
-        mimeType: file.mimetype,
-        fileContent: file.buffer,
-      },
-      studentId,
-      auditUserId,
-    );
-
-    return {
-      fileName: createdFile.fileName,
-      uniqueFileName: createdFile.uniqueFileName,
-      url: `student/files/${createdFile.uniqueFileName}`,
-      size: createdFile.fileContent.length,
-      mimetype: createdFile.mimeType,
-    };
+    let uploadStatus: number;
+    try {
+      this.logger.log(`Uploading file ${file.originalname} to S3 storage.`);
+      uploadStatus = await this.objectStorageService.putObject({
+        key: uniqueFileName,
+        contentType: file.mimetype,
+        body: file.buffer,
+      });
+    } catch (error: unknown) {
+      this.logger.error(parseJSONError(error));
+    }
+    if (uploadStatus === HttpStatus.OK) {
+      const createdFile = await this.fileService.createFile(
+        {
+          fileName: file.originalname,
+          uniqueFileName: uniqueFileName,
+          groupName: groupName,
+        },
+        studentId,
+        auditUserId,
+      );
+      if (createdFile) {
+        this.logger.log(`Uploaded file ${file.originalname} to S3 storage.`);
+        return {
+          fileName: createdFile.fileName,
+          uniqueFileName: createdFile.uniqueFileName,
+          url: `student/files/${createdFile.uniqueFileName}`,
+          size: file.size,
+          mimetype: file.mimetype,
+        };
+      } else {
+        // If the file details persistence to the database fails,
+        // remove the file from the s3 storage.
+        await this.objectStorageService.deleteObject(uniqueFileName);
```

Comment:
I did not anticipate this as part of the requirement.
Was it added because if we try to upload the same file twice it will generate an error on the S3?

Reply:
From my understanding, the form generates a `uniqueFileName` for each file during the upload process. Therefore, that should not be a problem.

<img width="651" alt="image" src="https://github.com/user-attachments/assets/9bb6285e-ab49-40fa-a8e4-6624e5454d8c">

However, I believe that if the saving of the file contents to our database failed, there is no point in keeping the file contents of that file in s3. That is why I was removing the file contents from s3 in case of unsuccessful save to our db.

Reply:
Removed the `deleteObject` method.